src/ServiceInsight.Tests/MessageListViewModelTests.cs

```diff
@@ -28,39 +29,40 @@ public class MessageListViewModelTests
         public void TestInitialize()
         {
             eventAggregator = Substitute.For<IEventAggregator>();
+            workNotifier = Substitute.For<IWorkNotifier>();
             serviceControl = Substitute.For<IServiceControl>();
             searchBar = Substitute.For<SearchBarViewModel>();
             clipboard = Substitute.For<IClipboard>();
-            messageListFunc = () => new MessageListViewModel(eventAggregator,
-                                                   serviceControl,
-                                                   searchBar,
-                                                   Substitute.For<GeneralHeaderViewModel>(),
-                                                   Substitute.For<MessageSelectionContext>(),
-                                                   clipboard);
+            messageListFunc = () => new MessageListViewModel(
+                eventAggregator,
+                workNotifier,
+                serviceControl,
+                searchBar,
+                Substitute.For<GeneralHeaderViewModel>(),
+                Substitute.For<MessageSelectionContext>(),
+                clipboard);
         }
 
         [Test]
         public void Should_load_the_messages_from_the_endpoint()
         {
             var endpoint = new Endpoint { Host = "localhost", Name = "Service" };
             serviceControl.GetAuditMessages(Arg.Is(endpoint), Arg.Any<string>(), Arg.Any<int>(), Arg.Any<string>(), Arg.Any<bool>())
-                             .Returns(x => new PagedResult<StoredMessage>
-                             {
-                                 CurrentPage = 1,
-                                 TotalCount = 100,
-                                 Result = new List<StoredMessage>
-                                 {
-                                     new StoredMessage(),
-                                     new StoredMessage()
-                                 }
-                             });
+                .Returns(x => new PagedResult<StoredMessage>
+                {
+                    CurrentPage = 1,
+                    TotalCount = 100,
+                    Result = new List<StoredMessage>
+                    {
+                        new StoredMessage(),
+                        new StoredMessage()
+                    }
+                });
 
             var messageList = messageListFunc();
 
             messageList.Handle(new SelectedExplorerItemChanged(new AuditEndpointExplorerItem(endpoint)));
 
-            eventAggregator.Received(1).PublishOnUIThread(Arg.Any<WorkStarted>());
-            eventAggregator.Received(1).PublishOnUIThread(Arg.Any<WorkFinished>());
```

Comment:
Should we preserve these assertions?

I [gave it a try](https://github.com/adamralph/ServiceInsight/commit/9356355b13826d8aeb32c4444504bd8e0fbedca7) and the test passes. The assertions on `workNotifier` and `notifyOfWork` are the equivalent of these two lines.

I'm not sure how important it is to test this interaction. I usually favour testing at the black box level first and drilling deeper only when required, but I don't know if this is already covered by any higher level test or whether it's practical to do so.